src/tools/w3dviewer/graphicview.cpp

```diff
@@ -0,0 +1,1095 @@
+    float f9 = 0.0f;
+    float f10 = 0.0f;
+    float f11 = f6 + f5;
+    float f12 = 0.0f;
+    float f13 = f1 * acceleration.Z * 0.5f + f7;
+    float f14;
+
+    if (acceleration.X == 0.0f) {
+        if (acceleration.Y == 0.0f && acceleration.Z == 0.0f) {
+            goto l1;
+        }
+
+        f14 = 0.0f;
+    } else {
+        f14 = -s.Center.X / acceleration.X;
+    }
+
+    float f15;
+
+    if (acceleration.Y == 0.0f) {
+        f15 = 0.0f;
+    } else {
+        f15 = -s.Center.Y / acceleration.Y;
+    }
+
+    float f16;
+
+    if (acceleration.Z == 0.0f) {
+        f16 = 0.0f;
+    } else {
+        f16 = -s.Center.Z / acceleration.Z;
+    }
+
+    if (f14 >= 0.0f && f14 < lifetime) {
+        f9 = GameMath::Fabs(f14 * f14 * acceleration.X * 0.5f + f14 * s.Center.X);
+    }
+
+    if (f15 >= 0.0f && f15 < lifetime) {
+        f12 = GameMath::Fabs(f15 * f15 * acceleration.Y * 0.5f + f15 * s.Center.Y);
+    }
+
+    if (f16 >= 0.0f && f16 < lifetime) {
+        f10 = GameMath::Fabs(f16 * f16 * acceleration.Z * 0.5f + f16 * s.Center.Z);
+    }
+
+l1:
+    float f17 = GameMath::Fabs(f8);
+    float f18 = GameMath::Fabs(f11);
+    float f19 = GameMath::Fabs(f13);
+    float f20;
+
+    if (f17 <= f18) {
+        f20 = f18;
+    } else {
+        f20 = f17;
+    }
+
+    if (f20 <= f19) {
+        f20 = f19;
+    }
+
+    if (f20 <= f9) {
+        f20 = f9;
+    }
+
+    if (f20 <= f12) {
+        f20 = f12;
+    }
+
+    if (f20 <= f10) {
+        f20 = f10;
+    }
+
+    float f21 = f17 * 0.5f;
+    s.Center.Y = f18 * 0.5f;
+    s.Center.Z = f19 * 0.5f;
+
+    float f24 = f9 * 0.5f;
+
+    if (f21 <= f24) {
+        f21 = f24;
+    }
+
+    float f25 = f12 * 0.5f;
+
+    if (s.Center.Y > f25) {
+        f25 = s.Center.Y;
+    }
+
+    float f26 = f10 * 0.5f;
+
+    if (s.Center.Z > f26) {
+        f26 = s.Center.Z;
+    }
+
+    s.Center.X = f21;
+    s.Center.Y = f25;
+    s.Center.Z = f26;
+    float f27 = f20 * 3.0f * 0.2f;
+    float f28 = buffer->Get_Particle_Size() * 5.0f;
+
+    if (f28 <= f27) {
+        s.Radius = f27;
+    } else {
+        s.Radius = f28;
+    }
+
+    ResetCameraValues(s);
+}
+
+void CGraphicView::SetRotationFlags(int flags)
+{
+    if (m_objectRotationFlags != flags) {
+        m_objectRotationFlags = flags;
+    }
+}
+
+void CGraphicView::SetCameraRotateConstraints(CameraRotateConstraint constraints)
+{
+    m_cameraRotateConstraints = constraints;
+}
+
+void CGraphicView::ResetRenderObj()
+{
+    CW3DViewDoc *document = GetCurrentDocument();
+
+    if (document != nullptr) {
+        if (document->m_model != nullptr) {
+            Matrix3D tm(true);
+            document->m_model->Set_Transform(tm);
+        }
+    }
+}
+
+void CGraphicView::UpdateCameraDistance(float distance)
+{
+    m_radius = distance;
+    Matrix3D tm(true);
+    tm.Look_At(
+        Vector3(distance + m_objSphere.Center.X, m_objSphere.Center.Y, m_objSphere.Center.Z), m_objSphere.Center, 0.0f);

```

Comment:
Can take out the `Vector3(distance + m_objSphere.Center.X, m_objSphere.Center.Y, m_objSphere.Center.Z)` in separate line to improve readability.

Reply:
fixed.